python/lsst/drp/tasks/assemble_cell_coadd.py

```diff
@@ -751,7 +751,11 @@ def run(
                 psf_shape_flag = True
                 psf_eval_point = None
                 try:
-                    if overlap_fraction < 1.0:
+                    # The `if` branch is buggy. `dest_polygon` is technically
+                    # out of scope, but Python does not raise an error.
+                    # TODO: Fix this properly in DM-53479, but sweep it under
+                    # the rug for now.
```

Comment:
Python's `for` scoping rules just don't work like those in other languages.

I'm guessing `overlap_fraction < 0.5` basically assumes this never fires?